homeassistant/components/mazda/sensor.py

```diff
@@ -1,21 +1,165 @@
 """Platform for Mazda sensor integration."""
 from __future__ import annotations
 
-from homeassistant.components.sensor import SensorEntity
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
+    CONF_UNIT_SYSTEM_IMPERIAL,
     LENGTH_KILOMETERS,
     LENGTH_MILES,
     PERCENTAGE,
     PRESSURE_PSI,
 )
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.util.unit_system import UnitSystem
 
 from . import MazdaEntity
 from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN
 
 
+@dataclass
+class MazdaSensorEntityDescription(SensorEntityDescription):
+    """Describes a Mazda sensor entity."""
+
+    # Suffix to be appended to the vehicle name to obtain the sensor name
+    name_suffix: str | None = None
```

Comment:
Why is name_suffix optional? All sensors provide a suffix so it should be required.
You may need to use a "Mixin" for the required keys. 
See https://github.com/home-assistant/core/blob/5fa6ef2849a9412e97526c98187ea68ed8b204f8/homeassistant/components/renault/sensor.py#L46-L65

Reply:
Thanks for the tip about the mixin. I originally made it optional because I thought I couldn't make it required. But I see that the mixin allows me to work around that. I fixed this, and also made `value` required because every sensor needs to provide a value function.